### PR TITLE
Updated headers and index for better flow and reference

### PR DESCRIPTION
**Pull request by cam586**
Index updated to flow better in the contents page
Headers in the files have been updated to reflect this

### CODE REVIEW

File: exotica/doc/index.rst [hunk 1]
```diff
@@ -9,16 +9,17 @@ EXOTica
...
--  `Home <index.html>`__
 -  `Installation <Installation.html>`_
--  `Requirements <Prerequisites.html>`__
--  `Quick Start: Python <Quickstart.html>`__
+-  `Quick Start: Python <Quickstart-python.html>`__
 -  `Quick Start: C++ <Quickstart-cpp.html>`__
 -  `Setting up Problems and Solvers <Setting-up-problems-and-solvers.html>`__
 -  `Using EXOTica <Using-EXOTica.html>`_ 
+
+-  `Requirements <Prerequisites.html>`__
```
Comment (wxmerkt) [inline exotica/doc/index.rst]: I'd like this header renamed - Requirements is way too generic and I had to open it to know what it means.

File: exotica/doc/index.rst [hunk 2]
```diff
@@ -32,28 +33,51 @@ Overview
...
+Advanced
+========
+
+With the advanced section, we look at different methods of initializing
+EXOTica and using it with your own robot. 
+
+.. toctree::
+   :maxdepth: 2
+
    Prerequisites
```
Comment (wxmerkt) [inline exotica/doc/index.rst]: Prerequisites has some broken links

File: exotica/doc/index.rst [hunk 3]
```diff
@@ -32,28 +33,51 @@ Overview
...
-   Quickstart
-   Quickstart-cpp
-   Setting-up-problems-and-solvers
-   Task_maps
    Setting-up-ROSlaunch
-   Using-EXOTica
-   Python-API
-   Documentation
+   Task_maps
+   Initializers
```
Comment (wxmerkt) [inline exotica/doc/index.rst]: Whis has this been moved so low if manual initialization is above? Perhaps group task maps and initializers under core components / concepts

File: exotica/doc/Installation.rst [hunk 4]
```diff
@@ -23,6 +23,25 @@ Installation
 5. Compile the code ``catkin build -s``.
 6. Source the config file (ideally inside ``~/.bashrc``): ``source path_to_workspace/devel/setup.bash``. You may have to source the config file from your install-space if your workspace is configured for installation.
 
+CMakeLists.txt & package.xml
```
Comment (wxmerkt) [inline exotica/doc/Installation.rst]: I disagree with moving this section to Installation, it's part of getting started or usage.

Comment (cam586) [reply exotica/doc/Installation.rst]: Yes, you're correct; I changed it over and altered the header title

**Pull request merged**